models/model.js

```diff
@@ -6,6 +6,29 @@ exports.selectTopics = () => {
   });
 };
 
+exports.selectArticles = (topic) => {
+  let topicStr = ``;
+  let queryValues = [];
+
+  if (topic) {
+    topicStr = `WHERE topic = $1`;
+    queryValues.push(topic);
+  }
+
+  let queryStr = `SELECT articles.*, COUNT(comments.comment_id) AS comment_count FROM articles LEFT JOIN comments ON comments.article_id = articles.article_id ${topicStr} GROUP BY articles.article_id ORDER BY created_at DESC;`;
+
+  return db.query(queryStr, queryValues).then(({ rows }) => {
+    if (rows.length === 0) {
```

Comment:
You will need to distinguish whether this is due to a non-existent topic or a topic with no articles in order to pass the suggested test.